zio/detector/reader.go

```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/brimsec/zq/zqe"
 )
 
-func NewReader(r io.Reader, zctx *resolver.Context) (zbuf.Reader, error) {
+func NewReader(r io.Reader, zctx *resolver.Context, path string, cfg OpenConfig) (zbuf.Reader, error) {
```

Comment:
I would prefer not to add more arguments to this simple interface. If this is necessary I would prefer just adding a new function: `NewReaderWithConfig`.

Separately what if OpenConfig.Format is set to `zng`. Does it make sense for it to return a json reader here?

Reply:
Even better than adding `NewReadingWithConfig` would be to just do type inference in whatever method invokes this:

```
if ndr, ok := r.(*ndjsonio.Reader); ok {
    ndr.SetConfig(path, cfg)
}
```

The scope of package `detector` should really limited to detecting the type of unknown streams. We've been adding a lot of bloat to this (I'm certainly guilty of this).

Reply:
> If this is necessary I would prefer just adding a new function: NewReaderWithConfig.

I was expecting that... sure I can use this color if you prefer.

Reply:
> Even better than adding NewReadingWithConfig would be to just do type inference in whatever method invokes this:
> `...`

I don't understand. What you're suggesting seems like exactly what `OpenFromNamedReadCloser` is already doing, right after calling NewReader. If I'm misunderstanding this, could you please clarify?

Reply:
> Separately what if OpenConfig.Format is set to zng. Does it make sense for it to return a json reader here?

It would not. But it would make even less sense to call this function with OpenConfig.Format that is set to a format. (Such an misplaced call to detector.NewReader() has always been possible, btw).

I could add a check for this, or change the interface to take a `*JSONTypeConfig` rather than the entire OpenConfig, if you like that better.